pineappl_capi/src/lib.rs

```diff
@@ -501,29 +503,30 @@ pub unsafe extern "C" fn pineappl_grid_convolve_with_one(
 /// # Safety
 ///
 /// If `grid` does not point to a valid `Grid` object, for example when `grid` is the null pointer,
-/// this function is not safe to call. The function pointers `xfx1`, `xfx2`, and `alphas` must not
+/// this function is not safe to call. The function pointers `xfx` and `alphas` must not
 /// be null pointers and point to valid functions. The parameters `order_mask` and `channel_mask`
 /// must either be null pointers or point to arrays that are as long as `grid` has orders and
 /// channels, respectively. Finally, `results` must be as long as `grid` has bins.
 #[no_mangle]
 pub unsafe extern "C" fn pineappl_grid_convolve_with_two(
     grid: *const Grid,
     pdg_id1: i32,
-    xfx1: extern "C" fn(pdg_id: i32, x: f64, q2: f64, state: *mut c_void) -> f64,
     pdg_id2: i32,
-    xfx2: extern "C" fn(pdg_id: i32, x: f64, q2: f64, state: *mut c_void) -> f64,
+    xfx: extern "C" fn(pdg_id: i32, x: f64, q2: f64, state: *mut c_void) -> f64,
     alphas: extern "C" fn(q2: f64, state: *mut c_void) -> f64,
-    state: *mut c_void,
+    pdf1_state: *mut c_void,
+    pdf2_state: *mut c_void,
+    alphas_states: *mut c_void,
```

Comment:
```suggestion
    alphas_state: *mut c_void,
```
why is there a plural? this seems wrong - `alphas` is bound to a fixed object

Reply:
actually in `pineappl_grid_convolve` it is a singular - so I assume it is just a typo here

Reply:
> why is there a plural? this seems wrong - `alphas` is bound to a fixed object

Good catch! That definitely has to be singular.

> actually in pineappl_grid_convolve it is a singular - so I assume it is just a typo here

As for `pineappl_grid_convolve`, the `convolve` part is a verb actually so that should be fine.